src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp

```diff
@@ -1958,8 +1955,214 @@ void jit_soft_sign_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, con
 }
 
 void jit_soft_sign_emitter::register_table_entries() {
-    push_arg_entry_of("one", 0x3f800000, true);
+    push_arg_entry_of("one", const_1F, true);
     push_arg_entry_of("positive_mask", 0x7fffffff, true);
 }
+
+/// IS_FINITE ///
+template <>
+void jit_is_finite_emitter::emit_isa<x64::avx512_core>(const std::vector<size_t> &inVecIdxs, const std::vector<size_t> &outVecIdxs) const {
+    auto vSrc = Zmm(inVecIdxs[0]);
+    auto vDst = Zmm(outVecIdxs[0]);
+    auto &onesMask = h->k1;
+    auto rOne = Reg32(aux_gpr_idxs[0]);
+
+    h->mov(rOne, const_1F);
+    h->vfpclassps(onesMask, vSrc, 0B10111001);
+    h->knotd(onesMask, onesMask);
+    h->vpbroadcastd(vDst | onesMask | h->T_z, rOne);
+}
+
+template <x64::cpu_isa_t isa>
+void jit_is_finite_emitter::emit_isa(const std::vector<size_t> &inVecIdxs, const std::vector<size_t> &outVecIdxs) const {
+    using Vmm = typename conditional<isa == x64::sse41, Xmm, Ymm>::type;
+    auto vSrc = Vmm(inVecIdxs[0]);
+    auto vDst = Vmm(outVecIdxs[0]);
+    auto vAux = Vmm(aux_vec_idxs[0]);
+    auto xmmAux = Xmm(vAux.getIdx());
+    auto rAux = Reg32(aux_gpr_idxs[0]);
+
+    h->mov(rAux, 0x7F800000); // Inf mask
+    h->uni_vpinsrd(xmmAux, xmmAux, rAux, 0x0);
+    h->uni_vpbroadcastd(vAux, xmmAux);
+    if (isa == x64::avx2) {
+        h->uni_vandps(vDst, vSrc, vAux);
+    } else {
+        h->uni_vmovups(vDst, vSrc);
+        h->uni_vandps(vDst, vDst, vAux);
+    }
+    h->uni_vcmpps(vDst, vDst, vAux, 0B00000100); // NEq
+
+    h->mov(rAux, 0x7FC00000); // NaN mask
+    h->uni_vpinsrd(xmmAux, xmmAux, rAux, 0x0);
+    h->uni_vpbroadcastd(vAux, xmmAux);
+    h->uni_vandps(vSrc, vSrc, vAux);
+    h->uni_vcmpps(vSrc, vSrc, vAux, 0B00000100); // NEq
+    h->uni_vandps(vDst, vDst, vSrc);
+
+    h->mov(rAux, const_1F);
+    h->uni_vpinsrd(xmmAux, xmmAux, rAux, 0x0);
+    h->uni_vpbroadcastd(vAux, xmmAux);
+    h->uni_vandps(vDst, vDst, vAux);
+}
+
+void jit_is_finite_emitter::emit_impl(const std::vector<size_t> &inVecIdxs, const std::vector<size_t> &outVecIdxs,
+                                      const std::vector<size_t> &poolVecIdxs, const std::vector<size_t> &poolGprIdxs,
+                                      const emitter_context *emitContext) const {
+    if (host_isa_ == x64::avx512_core) {
+        emit_isa<x64::avx512_core>(inVecIdxs, outVecIdxs);
+    } else if (host_isa_ == x64::avx2) {
+        emit_isa<x64::avx2>(inVecIdxs, outVecIdxs);
+    } else if (host_isa_ == x64::sse41) {
+        emit_isa<x64::sse41>(inVecIdxs, outVecIdxs);
+    } else {
+        assert(!"Unsupported isa");
```

Comment:
I'd suggest to throw IE exception with more informative message, something like this:
`IE_THROW() << "jit_is_finite_emitter doesn't support " << host_isa_;`

Reply:
Done